File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProfit = 0
        curBuy = prices[0]
        for p in prices:
            if p < curBuy:
                curBuy = p
            profit = p - curBuy
            maxProfit = max(maxProfit, profit)
        return maxProfit

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        if (n <= 0) return 0;
        int mProfit = 0, buyPrice = prices[0];
        for (int i = 0; i < n; i++) {
            if (buyPrice < prices[i]) {
                mProfit += prices[i] - buyPrice;
            }
            buyPrice = prices[i];
        }
        return mProfit;
    }
};